core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java

```diff
@@ -18,843 +18,73 @@
-      }
-
-      @Override
-      public OutputStream createCompressionStream(OutputStream downStream, Compressor compressor,
-          int downStreamBufferSize) throws IOException {
-        if (!isSupported()) {
-          throw new IOException(
-              "ZStandard codec class not specified. Did you forget to set property "
-                  + CONF_ZSTD_CLASS + "?");
-        }
-        return createFinishedOnFlushCompressionStream(downStream, compressor, downStreamBufferSize);
-      }
-
-      @Override
-      public InputStream createDecompressionStream(InputStream downStream,
-          Decompressor decompressor, int downStreamBufferSize) throws IOException {
-        if (!isSupported()) {
-          throw new IOException(
-              "ZStandard codec class not specified. Did you forget to set property "
-                  + CONF_ZSTD_CLASS + "?");
-        }
-        return createDecompressionStream(downStream, decompressor, downStreamBufferSize,
-            DEFAULT_BUFFER_SIZE, ZSTANDARD, codec);
-      }
-
-      @Override
-      public boolean isSupported() {
-        return codec != null;
-      }
-    };
-
-    /**
-     * Guava cache to have a limited factory pattern defined in the Algorithm enum.
-     */
-    private static LoadingCache<Entry<Algorithm,Integer>,CompressionCodec> codecCache =
-        CacheBuilder.newBuilder().maximumSize(25).build(new CacheLoader<>() {
-          @Override
-          public CompressionCodec load(Entry<Algorithm,Integer> key) {
-            return key.getKey().createNewCodec(key.getValue());
-          }
-        });
+  private static final Map<String,DefaultCompressionAlgorithm> CONFIGURED_ALGORITHMS =
+      new HashMap<>();
+
+  private static final ServiceLoader<CompressionAlgorithm> COMPRESSION_ALGORITHMS =
+      ServiceLoader.load(CompressionAlgorithm.class);
+
+  // All compression-related settings are required to be configured statically in the
+  // Configuration object.
+  protected static final Configuration conf;
+
+  // The model defined by the static block below creates a singleton for each defined codec in the
+  // Algorithm enumeration. By creating the codecs, each call to isSupported shall return
+  // true/false depending on if the codec singleton is defined. The static initializer, below,
+  // will ensure this occurs when the Enumeration is loaded. Furthermore, calls to getCodec will
+  // return the singleton, whether it is null or not.
+  //
+  // Calls to createCompressionStream and createDecompressionStream may return a different codec
+  // than getCodec, if the incoming downStreamBufferSize is different than the default. In such a
+  // case, we will place the resulting codec into the codecCache, defined below, to ensure we have
+  // cache codecs.
+  //
+  // Since codecs are immutable, there is no concern about concurrent access to the
+  // CompressionCodec objects within the guava cache.
+  static {
+    conf = new Configuration();
```

Comment:
I don't know if we still need this static code block. This was probably written before we had ServerContext. I feel like we should just be getting the configuration from ServerContext.

Reply:
The static block is still needed for the other code in it, right?

Reply:
> I feel like we should just be getting the configuration from ServerContext.

Can't. This is in `core` with the rest of the file-writing code. Maybe ClientContext, though.

> The static block is still needed for the other code in it, right?

No. See my other comment.